src/distibutions/v8-canary/canary_builds.ts

```diff
@@ -0,0 +1,94 @@
+import * as core from '@actions/core';
+import * as tc from '@actions/tool-cache';
+
+import semver from 'semver';
+
+import BaseDistribution from '../base-distribution';
+import {INodejs} from '../base-models';
+
+export default class CanaryBuild extends BaseDistribution {
+  protected distribution = 'v8-canary';
+  constructor(nodeInfo: INodejs) {
+    super(nodeInfo);
+  }
+
+  protected findVersionInHoostedToolCacheDirectory(): string {
```

Comment:
This method seems to be identical in all subclasses. Can we move it to base class? Or may be simplify it somehow to avoid code duplication.